src/MongoDB.Driver/Linq/FieldExpressionFlattener.cs

```diff
@@ -86,5 +80,43 @@ protected internal override Expression VisitField(FieldExpression node)
 
             return node;
         }
+
+        // private methods
+        private static string FormatArrayIndex(object index)
+        {
+            if (index is int intIndex)
+            {
+                return GetFormattedIndexOrThrow(intIndex);
```

Comment:
I guess here we can push `int` to `long` parameter without any disadvantages.

Reply:
Agreed

Reply:
Let's put in an explicit cast just to highlight what's happening:

    return FormatArrayIndex((long)intIndex);

Reply:
Done